src/EditorFeatures/Core/Extensibility/Navigation/NavigableItemFactory.cs

```diff
@@ -59,6 +59,16 @@ private static IEnumerable<Location> GetPreferredSourceLocations(ISymbol symbol)
                 : locations.Where(loc => loc.IsInSource);
         }
 
+        public static IEnumerable<INavigableItem> GetPreferredNavigableItems(Solution solution, IEnumerable<INavigableItem> navigableItems)
+        {
+            var generatedCodeRecognitionService = solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>();
+            navigableItems = navigableItems.Where(n => n.Document != null);
+            var hasNonGeneratedCodeItem = navigableItems.Any(n => !generatedCodeRecognitionService.IsGeneratedCode(n.Document));
+            return hasNonGeneratedCodeItem ?
+                navigableItems.Where(n => !generatedCodeRecognitionService.IsGeneratedCode(n.Document)) :
+                navigableItems.Where(n => generatedCodeRecognitionService.IsGeneratedCode(n.Document));
```

Comment:
I don't like this.  Currently we filter out results from generated code, and it makes the experience in Roslyn itself very unpleasant.  Specifically, trying to navigate to syntax node types is super hard to do.  I don't seewhy we want to filter things out.